packages/langium/src/workspace/ast-descriptions.ts

```diff
@@ -53,15 +52,15 @@ export class DefaultAstNodeDescriptionProvider implements AstNodeDescriptionProv
         if (!name) {
             throw new Error(`Node at path ${path} has no name.`);
         }
-        let nameNodeSegment: DocumentSegment | undefined;
-        const nameSegmentGetter = () => nameNodeSegment ??= toDocumentSegment(this.nameProvider.getNameNode(node) ?? node.$cstNode);
+        const nameProperty = this.nameProvider.getNameProperty(node);
+        const nameSegment: DocumentSegment | undefined = nameProperty
+            ? node.$segments?.properties.get(nameProperty)[0]
+            : undefined;
```

Comment:
The description providers are rather central services. I suggest to keep the CST-based segment resolution as fallback.